main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -1485,6 +1477,11 @@ object GenExpression {
 
   }
 
+  private def visitArrayInstantiate(mv: MethodVisitor, tpe: BackendType): Unit = tpe match {
+    case BackendType.Array(_) | BackendType.Reference(_) => mv.visitTypeInsn(ANEWARRAY, AsmOps.getArrayType(tpe))
+    case _ => mv.visitIntInsn(NEWARRAY, AsmOps.getArrayTypeCode(tpe).head)
```

Comment:
If `getArrayTypeCode` is only used here, then I think it should be inlined. Then you avoid the option.